pkg/agent/flowexporter/connections/connections.go

```diff
@@ -115,6 +115,10 @@ func (cs *connectionStore) Run(stopCh <-chan struct{}) {
 // addOrUpdateConn updates the connection if it is already present, i.e., update timestamp, counters etc.,
 // or adds a new Connection by 5-tuple of the flow along with local Pod and PodNameSpace.
 func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
+	// avoid adding or updating the connection that is about to close
+	if conn.IsExportedAndDying {
```

Comment:
This will not work as it is the connection from the dump after polling. Here the unit test for the below-mentioned scenario will help.